driver.py

```diff
@@ -10,14 +14,45 @@ def run_driver():
 
     engine.activate('rules')
 
-    print("doing proof")
+    print()
+    print('----------------------')
+    print("Welcome to Beer-Me-Up!")
+    print('----------------------')
 
     try:
+        beers = []
         with engine.prove_goal(
-                'rules.beer($beer)') as gen:  # STUDENTS: you will need to edit this line
+                'rules.beer($beer, $link, $explanation)') as gen:
+
+            # Get beers
+            i = 0
             for vars, plan in gen:
-                # STUDENTS: you will need to edit this line
-                print("You should drink: %s" % (vars['beer']))
+                i = i + 1
+                beer = Beer(vars['beer'], vars['link'], vars['explanation'])
+                beers.append(beer)
+
+        # Get unique beers
+        keys = list(set(beers))
+        print('\nGenerating result...\n')
+        print(f'{len(keys)} beers found.')
+
+        # Create dictionary
+        dictionary = {}
+        for beer in beers:
+            if dictionary.get(beer.name) is not None:
+                dictionary[beer.name].add(beer.explanation)
+            else:
+                dictionary[beer.name] = BeerInfo(beer)
```

Comment:
I czy dobrze myślę, że w przypadku kiedy piwa nie ma w słowniku jeszcze, czyli trafiamy na jego pierwsze wystapienie w liście beers to on bedzie miał tu count ustawiony na 0? Chyba powinnismy w BeerInfo count inicjalizować 1

Reply:
faktycznie powinno być 1, poprawię